aspnetcore/performance/rate-limit.md

```diff
@@ -19,6 +19,150 @@ The `Microsoft.AspNetCore.RateLimiting` middleware provides rate limiting middle
+{
+    options.GlobalLimiter = PartitionedRateLimiter.Create<HttpContext, string>(httpContext =>
+        RateLimitPartition.GetFixedWindowLimiter(
+            partitionKey: httpContext.User.Identity?.Name ?? httpContext.Request.Headers.Host.ToString(),
+            factory: partition => new FixedWindowRateLimiterOptions
+            {
+                AutoReplenishment = true,
+                PermitLimit = 10,
+                QueueLimit = 0,
+                Window = TimeSpan.FromMinutes(1)
+            }));
+});
+```
+
+Or as named polices, which need to be explicitly applied to the pages or endpoints. For example, to add a fixed window limiter:
+
+``` csharp
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddRateLimiter(options =>
+{
+    options.AddFixedWindowLimiter("fixed", opt =>
+    {
+        opt.PermitLimit = 4;
+        opt.Window = TimeSpan.FromSeconds(12);
+        opt.QueueProcessingOrder = QueueProcessingOrder.OldestFirst;
+        opt.QueueLimit = 2;
+    });
+});
+
+var app = builder.Build();
+```
+
+The global limiter applies to all endpoints automatically when it's configured via options.GlobalLimiter, and no endpoint-specific policy is specified.
+
+### 3. Enable rate limiting middleware
+ In the Program.cs file, enable the rate limiting middleware by calling UseRateLimiter:
+
+``` csharp
+app.UseRouting();
+
+app.UseRateLimiter();
+
+app.UseEndpoints(endpoints =>
+{
+    endpoints.MapControllers();
+});
+
+app.Run();
+```
+
+### Apply rate limiting policies to endpoints or pages
+
+#### Apply rate limiting to WebAPI Endpoints
+
+Apply a named policy to the endpoint or group, for example:
+
+``` csharp
+
+app.MapGet("/api/resource", () => "This endpoint is rate limited")
+   .RequireRateLimiting("fixed"); // Apply specific policy to an endpoint
+
+```
+
+#### Apply rate limiting to MVC Controllers
+
+ Apply the configured rate limiting policies to specific endpoints or globally. For example, to apply the "fixed" policy to all controller endpoints:
+
+``` csharp
+app.UseEndpoints(endpoints =>
+{
+    endpoints.MapControllers().RequireRateLimiting("fixed");
+});
+
+```
+
+#### Apply rate limiting to Blazor Server Pages
+
+To set rate limiting to all pages,  `RequireRateLimiting(Policy)` can be specified on the MapRazorComponents call, for example:
+
+``` csharp
+app.MapRazorComponents<App>()
+    .AddInteractiveServerRenderMode()
+    .RequireRateLimiting("MyPagePolicy");
+```
+
+To set policy for individual Blazor Pages, the attribute must be applied to the Page and not the page handlers.
+
+``` blazor
+@page "/counter"
+@using Microsoft.AspNetCore.RateLimiting
+@attribute [EnableRateLimiting("blazor")]
+
+<h1>Counter</h1>
+```
+
+The `DisableRateLimiting` attribute can be used to disable rate limiting on a Razor Page.
+
+Note: `EnableRateLimiting` is only applied to a Razor Page if `MapBlazorComponents().RequireRateLimiting(Policy)` has ***not*** been called.
```

Comment:
This seems out of position: I recommend moving it up to where the attribute is described. Either way on that, the following polishes it up a bit ...

```suggestion
The [`[EnableRateLimiting]` attribute](xref:Microsoft.AspNetCore.RateLimiting.EnableRateLimitingAttribute) is only applied to a component if <xref:Microsoft.AspNetCore.Builder.RateLimiterEndpointConventionBuilderExtensions.RequireRateLimiting%2A> is ***not*** called on <xref:Microsoft.AspNetCore.Builder.RazorComponentsEndpointRouteBuilderExtensions.MapRazorComponents%2A>.
```